src/Kdyby/DoctrineCache/Cache.php

```diff
@@ -96,12 +96,13 @@ protected function doSave($id, $data, $lifeTime = 0)
 				$files[] = ClassType::from($class)->getFileName();
 			}
 		}
+		if ($data instanceof \Symfony\Component\Validator\Mapping\ClassMetadata) {
+			$files[] = ClassType::from($data->name)->getFileName();
+		}
 
 		if (!empty($data)){
-			if ((is_array($data) && reset($data) instanceof Doctrine\ORM\Mapping\Annotation) || $data instanceof Doctrine\ORM\Mapping\Annotation) {
```

Comment:
> Is there anything apart from annotations, that is using the same cache id scheme?

Not that I know of but that doesn't mean anything...

My proposal about `DependenciesProviderInterface` would solve that though.